internal/services/kubernetesconfiguration/kubernetes_cluster_extension_resource.go

```diff
@@ -0,0 +1,385 @@
+}
+
+type PlanModel struct {
+	Name          string `tfschema:"name"`
+	Product       string `tfschema:"product"`
+	PromotionCode string `tfschema:"promotion_code"`
+	Publisher     string `tfschema:"publisher"`
+	Version       string `tfschema:"version"`
+}
+
+type KubernetesClusterExtensionResource struct{}
+
+var _ sdk.ResourceWithUpdate = KubernetesClusterExtensionResource{}
+
+func (r KubernetesClusterExtensionResource) ResourceType() string {
+	return "azurerm_kubernetes_cluster_extension"
+}
+
+func (r KubernetesClusterExtensionResource) ModelObject() interface{} {
+	return &KubernetesClusterExtensionModel{}
+}
+
+func (r KubernetesClusterExtensionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return extensions.ValidateExtensionID
+}
+
+func (r KubernetesClusterExtensionResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := commonArguments()
+	arguments["plan"] = &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		ForceNew: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"name": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"product": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"publisher": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"promotion_code": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"version": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+		},
+	}
+
+	return arguments
+}
+
+func (r KubernetesClusterExtensionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"identity": commonschema.SystemAssignedIdentityComputed(),
+
+		"current_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r KubernetesClusterExtensionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model KubernetesClusterExtensionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.KubernetesConfiguration.ExtensionsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := extensions.NewExtensionID(subscriptionId, model.ResourceGroupName, "Microsoft.ContainerService", "managedClusters", model.ClusterName, model.Name)
```

Comment:
```suggestion
			// defined as strings because they're not enums in the swagger https://github.com/Azure/azure-rest-api-specs/pull/23545
			id := extensions.NewExtensionID(subscriptionId, model.ResourceGroupName, "Microsoft.ContainerService", "managedClusters", model.ClusterName, model.Name)
```